tests/macsec/test_reboot.py

```diff
@@ -0,0 +1,105 @@
+'''This script is to test the device reboot behavior for SONiC.
+
+Test 1: Line Card
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the line card
+Step 3: Verify macsec connection is re-established after reboot
+
+Test 2: Chassis
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the chassis
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import logging
+import pytest
+import re
+import time
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+def get_macsec_sessions(dut, space_var):
```

Comment:
This function is meant to only collect sessions that are UP. However, since there is no proper filtering here I think it now will collect MACsec sessions that are down as well.

Reply:
I have adjusted this function to add a check to ensure the session is enabled.

Reply:
The filtering that you add can only verify that the session is enabled. You need to use something similar to check_mka_establishment() in PR #10719 to check if the session is UP.

Reply:
I am not understanding what you are asking for.  If a session is enabled then it is active on the link (at least based on our testing as if there is a problem the session is not enabled, but still listed).  The check_mka_establishment function does less than the check I added here, it only sees if one of three tables of information on the port exists.

Reply:
As long as the "enable" field in a port's macsec config is set to true, then the port is shown as enabled in "show macsec" command. However, that does not necessarily mean that the Macsec session of this port is UP. A macsec session of a port is UP if and only if it successfully established SC/SA with peers. I am not telling you to reuse check_mka_establishment function. You can use "show macsec" but gather SC/SA info instead of simply gathering "enable".

Reply:
I adjusted this check to look for the SA data that is only shown when the session is up.

Reply:
I assume that your lines are 0-indexed, since index 3 corresponds to "enable" line. Then, the line corresponding to "Egress SA" should be indexed by 14. Why do you check "count == 15"?

Reply:
I get 15 when I count the lines to that information.  The script only appends the session when it gets to that line and I looked at the debug to make sure the sessions were being added and they all are.

Reply:
I think this checking based on the assumption on how many lines there are before the first Egress SA is not robust. I think this applies to the checking for "enable" also. If the info printed by "show macsec" is reordered or expanded in the future, this will fail. Let's make it more robust.

How about change the checking of "enable" to the following:
```
elif "enable" in line.split() and "true" in line:
    en = True
```
change the checking of "Egress SA" to the following:
```
elif en and "MACsec Egress SA" in line:
    sess_list.append(temp_sess)
```

By the way, you should reset `en` when you find a new session.

Reply:
I have adjusted as per these comments.